src/formatDate.js

```diff
@@ -9,6 +9,54 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const oldDateParts = date.split(oldSeparator);
+
+  let datePartsMatched = {};
+
+  for (let i = 0; i < oldDateParts.length; i++) {
+    datePartsMatched[fromFormat[i]] = oldDateParts[i];
+  }
+
+  datePartsMatched = convertYear(datePartsMatched);
+
+  const newDateParts = [];
+
+  for (let i = 0; i < oldDateParts.length; i++) {
+    const newDateFormat = toFormat[i];
+
+    const newDatePart = datePartsMatched[newDateFormat];
+
+    newDateParts.push(newDatePart);
+  }
+
+  return newDateParts.join(newSeparator);
+}
+/**
+ * @param {Object} datePartsMatched
+ *
+ * @returns {Object}
+ */
+
+function convertYear(datePartsMatched) {
+  const FOUR_DIGIT_YEAR_FORMAT = 'YYYY';
+  const TWO_DIGIT_YEAR_FORMAT = 'YY';
+
+  if (FOUR_DIGIT_YEAR_FORMAT in datePartsMatched) {
+    const year = datePartsMatched[FOUR_DIGIT_YEAR_FORMAT];
+
+    datePartsMatched[TWO_DIGIT_YEAR_FORMAT] = year.slice(2);
```

Comment:
When converting from a four-digit year to a two-digit year, you're always assigning the value to the `YY` key, even if the `toFormat` does not include `YY`. This could lead to incorrect results if the `toFormat` array expects a four-digit year. You should check if the `toFormat` array contains `YY` before making this assignment.